frontend/mobilityai/app/src/main/java/com/ai/mobility/mobilityai/MainActivity.java

```diff
@@ -0,0 +1,324 @@
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+
+        initialize();
+
+        setUpBluetoothScanner();
+
+        // Bind the service when the activity is created
+        getApplicationContext().bindService(new Intent(this, BtleService.class),
+                this, Context.BIND_AUTO_CREATE);
+    }
+
+    //TODO: Delete this function once Bluetooth stuff is working
+    private String getRandomName() {
+        String[] names = {"Rebecca Tran", "Roberto Temelkovski", "Teo Voinea", "梓川 咲太"};
+        int rnd = r.nextInt(names.length);
+        return names[rnd];
+    }
+
+    @Override
+    public void onDestroy() {
+        stopBleScan();
+        super.onDestroy();
+
+        // Unbind the service when the activity is destroyed
+        getApplicationContext().unbindService(this);
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        switch (requestCode) {
+            case REQUEST_ENABLE_BT:
+                startBleScan();
+                break;
+        }
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        getMenuInflater().inflate(R.menu.menu_main, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Handle action bar item clicks here. The action bar will
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        int id = item.getItemId();
+
+        //noinspection SimplifiableIfStatement
+        if (id == R.id.action_settings) {
+            return true;
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    public void onServiceConnected(ComponentName name, IBinder service) {
+        serviceBinder = (BtleService.LocalBinder) service;
+
+        Log.i(TAG, "Service Connected");
+
+//        retrieveBoard();
+    }
+
+    @Override
+    public void onServiceDisconnected(ComponentName name) {
+        if(board != null) {
+            board.disconnectAsync().continueWith(new Continuation<Void, Void>() {
+                @Override
+                public Void then(Task<Void> task) throws Exception {
+                    return null;
+                }
+            });
+        }
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String permissions[], int[] grantResults) {
+        switch (requestCode) {
+            case PERMISSION_REQUEST_COARSE_LOCATION:
+                if(grantResults[0] != PackageManager.PERMISSION_GRANTED) {
+                    //TODO: Handle if permission denied
```

Comment:
Maybe show a popup that the app won't work without this permission? Could create an issue for that.

Reply:
Done, see #79